department-of-veterans-affairs/abd-vro
# Bgs liveness

## What was the problem?
Implementing K8s Liveness and Readiness on BGS-api (Ruby application)

Associated tickets or Slack threads:
- #2417 

## How does this fix it?[^1]
This configures K8s liveness and readiness for svc-bgs-api to ensure highest level of visibility and reliability into the health and readiness of platform and partner team services.

## How to test this PR
- Step 1
    - Use Lens to check Pod Descriptions for Readiness Probe Output
    
- Step 2
    - Simulate Probe Failure to test if the probes are working correctly. 
    - **For Liveness:** Disrupt connection between RabbitMQ and BGS-api 
    - **For Readiness:** Temporarily disrupt the connection to RabbitMQ and also disrupt BGS service and make it unavailable 
       or unresponsive. 
    - Check if BOTH liveness and readiness probes correctly identifies these. Pods will restart if Liveness probe fails while 
       pods will be marked as 'Not Ready' if Readiness probe fails.
